src/main/kotlin/org/opensearch/replication/ReplicationPlugin.kt

```diff
@@ -359,7 +363,13 @@ internal class ReplicationPlugin : Plugin(), ActionPlugin, PersistentTaskPlugin,
 
     override fun getEngineFactory(indexSettings: IndexSettings): Optional<EngineFactory> {
         return if (indexSettings.settings.get(REPLICATED_INDEX_SETTING.key) != null) {
-            Optional.of(EngineFactory { config -> ReplicationEngine(config) })
+            Optional.of(EngineFactory { config ->
+                if (indexSettings.settings.get(INDEX_REPLICATION_TYPE_SETTING.key).equals("SEGMENT")) {
+                    if (config.isReadOnlyReplica) NRTReplicationEngine(config) else ReplicationEngine(config)
```

Comment:
It should be `indexSettings.settings.get(INDEX_REPLICATION_TYPE_SETTING.key).equals("SEGMENT") && config.isReadOnlyReplica`

Reply:
Actually, we shouldn't be setting the Engine here. Instead, we can override the ReplicationEngine only if the shard is `primary`. This way the we don't replicate the logic and  existing `EngineFactory` would kick in add the `NRTReplciationEngine` for replicas.

Reply:
Just realised that primary shard check might not work as we load the `EngineFactory` as index creation level and not shard. 
We'll need to keep the logic to load the `NRTReplicationEngine` here.

Reply:
Updated the logic here.